Add docs for pipeline validation (#868)
elyra-ai/elyra: docs/source/user_guide/pipelines.md

At this time, the only recourse, should it not be possible to fix the relative location via redeployment of the pipeline, would be to edit each of the `filename:` fields in the `.pipeline` file such that notebook locations are satisfied, then refresh the browser.

## Pipeline Validation
Pipeline validation occurs when pipeline files are opened, as well as when pipelines are run or exported. Pipelines are validated for the following:
- **Circular References** - Circular references cannot exist in any pipeline because it would create an infinite loop. 

> - **Notebook Existence** - The notebook for a given node must able to be located. 

Reviewer: I think you might be missing a 'be' here - or perhaps something like:
```suggestion
- **Notebook Existence** - The notebook for a given node must exist. 
```
I wasn't sure if we should state anything about the notebook path being relative to the pipeline or the notebook-dir, and I think its best we not mention either.
